nodes/filter/nodes.go

```diff
@@ -0,0 +1,225 @@
+
+	if inputNull || matchNull {
+		inst.WritePinNull(node.Out)
+	} else {
+		if float.RoundTo(input, 2) != float.RoundTo(match, 2) {
+			inst.WritePinFloat(node.Out, input)
+			inst.lastValue = input
+		} else {
+			inst.WritePinFloat(node.Out, inst.lastValue)
+		}
+	}
+}
+
+// preventEqualString, output the last known value if two inputs equal to each other. Out put nil if any of the inputs are nil.
+// lastValue is defaulted to be "defaultString"
+func (inst *PreventEqualString) Process() {
+	input, inputNull := inst.ReadPinAsString(node.In)
+	match, matchNull := inst.ReadPinAsString(node.InputMatch)
+
+	if inputNull || matchNull {
+		inst.WritePinNull(node.Out)
+	} else {
+		if input != match {
+			inst.WritePin(node.Out, input)
+			inst.lastValue = input
+		} else {
+			inst.WritePin(node.Out, inst.lastValue)
+		}
+	}
+}
+
+// onlyBetween, Output the last known input float value if the input is not in range. Out put nil if any of the inputs are nil.
+// lastValue is defaulted to be 0
+func (inst *OnlyBetween) Process() {
+	input, inputNull := inst.ReadPinAsFloat(node.In)
+	min, minNull := inst.ReadPinAsFloat(node.InMin)
+	max, maxNull := inst.ReadPinAsFloat(node.InMax)
+
+	if inputNull || minNull || maxNull {
+		inst.WritePinNull(node.Out)
+	} else {
+		if float.RoundTo(min, 2) <= float.RoundTo(input, 2) && float.RoundTo(input, 2) <= float.RoundTo(max, 2) {
+			inst.WritePinFloat(node.Out, input)
+			inst.lastValue = input
+		} else {
+			inst.WritePinFloat(node.Out, inst.lastValue)
+		}
+	}
+}
+
+// onlyGreater, Output the last known input float value if the input is less than 'OutMax'. Out put nil if any of the inputs are nil.
+// lastValue is defaulted to be 0
+func (inst *OnlyGreater) Process() {
+	input, inputNull := inst.ReadPinAsFloat(node.In)
+	min, minNull := inst.ReadPinAsFloat(node.InMin)
+
+	if inputNull || minNull {
+		inst.WritePinNull(node.Out)
+	} else {
+		if float.RoundTo(input, 2) > float.RoundTo(min, 2) {
+			inst.WritePinFloat(node.Out, input)
+			inst.lastValue = input
+		} else {
+			inst.WritePinFloat(node.Out, inst.lastValue)
+		}
+	}
+}
+
+// onlyLower, Output the last known input float value if the input is greater than 'OutMin'. Out put nil if any of the inputs are nil.
+// lastValue is defaulted to be 0
+func (inst *OnlyLower) Process() {
+	input, inputNull := inst.ReadPinAsFloat(node.In)
+	max, maxNull := inst.ReadPinAsFloat(node.InMax)
+
+	if inputNull || maxNull {
+		inst.WritePinNull(node.Out)
+	} else {
+		if float.RoundTo(input, 2) < float.RoundTo(max, 2) {
+			inst.WritePinFloat(node.Out, input)
+			inst.lastValue = input
+		} else {
+			inst.WritePinFloat(node.Out, inst.lastValue)
+		}
+	}
+}
+
+// preventDuplicates, don't do anything if the subsequent input value is the same. Out put nil if the input is nil.
+// lastValue is defaulted to be 0
+func (inst *PreventDuplicates) Process() {
+	input, inputNull := inst.ReadPinAsFloat(node.In)
+
+	if inputNull {
```

Comment:
missed some logic here.  If the input is `null`, then it will send duplicate `null` on each loop.  If the input is `null` set `inst.lastValue` to `nil`, and do the same comparison you would do for floats. (you will need to update the type to `*float64` and handle the `nil` cases so it doesn't crash.